### ISSUE TITLE
Some recipes fail when plotting empty vectors

### ISSUE THREAD

**Issue by BambOoxX** [JuliaPlots/Makie.jl#3567]
When plotting empty objects, some recipes fail on `convert_arguments`, but not all of them do, making it difficult to know if it should be an expected behavior or not. 
I run into this problem mostly with `arrows` because sometimes I have missing objects in some model. The problem is that I do not know what is better. Not plotting at all (handling the lack of data at my level), or attempting plotting regardless of the underlying data, to preserve the plot structure. 

I there any recommendation regarding this ? 
This is somewhat related to #3228 

As an example, `scatter` works, but `arrows` do not. 

I've tested this for multiple recipes (listed below) mostly in 3D. 

```julia
ablines([],[]) # works
arrows(Point3f[], Vec3f[]) # fails - works now
scatter(Point3f[]) # works
band(Point3f[],Point3f[]) # fails
barplot(Float64[],Float64[]) # works 
boxplot(Float64[],Float64[]) # works
lines(Point3f[]) # works
linesegments(Point3f[]) # works
meshscatter(Point3f[]) # works
pie(Float64[]) # works
scatterlines(Point3f[]) # fails - works now
triplot(Point3f[]) # fails
```

- [x] are you running newest version (version from docs) ?
- [x] can you reproduce the bug with a fresh environment ? (`]activate --temp; add Makie`)
- [x] What platform + GPU are you on?
>julia> versioninfo()
Julia Version 1.10.0
Commit 3120989f39 (2023-12-25 18:01 UTC)
Build Info:
  Official https://julialang.org/ release
Platform Info:
  OS: Windows (x86_64-w64-mingw32)
  CPU: 8 × 11th Gen Intel(R) Core(TM) i7-1185G7 @ 3.00GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-15.0.7 (ORCJIT, tigerlake)
  Threads: 1 on 8 virtual cores

**Comment by jkrumbiegel** [comment]
Most plotting functions should handle empty vectors. I can't formulate a perfect rule but usually if you can draw zero or more of something, then this should work for zero. But if there's some data transformation inherent to a recipe which just doesn't make sense for zero elements, then this should fail. Like `sum(Float64[])` returns `0` but `mean(Float64[])` returns `NaN` because you can't divide by zero.

Often, problems with zero elements will stem from type instabilities breaking subsequent dispatches. For example, a vector will be assembled given the input data and this ends up `Any[]` for the zero-element case because it couldn't be inferred what element type it should be. Then any following dispatch will probably fail if it expected `Point2f` or so. These are often reasonably easy to fix by type-constraining the relevant containers.

**Comment by BambOoxX** [comment]
@jkrumbiegel Yes, that is why I explicitely called concretely typed empty vectors, that avoids the dispatch problem I guess, at least partially.

**Comment by jkrumbiegel** [comment]
Right but this is not just about the input types, the recipes themselves will sometimes do type-unstable transformations. But those can be fixed one by one. We should definitely increase the test coverage for empty recipes.

**Comment by BambOoxX** [comment]
@jkrumbiegel i hope I will have a bit of time for that but so far I've never contributed to make. I have to say I do not know how to dev it properly without breaking the rest of my environment. I know there is a section about contributions but I had surprises with previous attempts. Any suggestions?

**Comment by jkrumbiegel** [comment]
Right, it's a little more complex than with other packages but not terribly so. I have a shared `@makie` environment that I activate when developing, because if you just activate the main `Makie` env directly, you'll not get `MakieCore`, and any of the backends dev'ed.

<img width="448" alt="image" src="https://github.com/MakieOrg/Makie.jl/assets/22495855/9e02697d-1505-4898-8b8e-82cc55b197d5">

So `]activate @makie` and then `dev Makie` should clone the whole Makie repo into your dev folder. Then you can `dev MakieCore` etc and it should find those in the Makie folder because they are subfolders of that.

**Comment by ffreyer** [comment]
updated the opening post for what's not erroring. 2 to go

**Comment by mikmoore** [comment]
On Makie `v0.24.2`/ CairoMakie `v0.15.2` there appear to be a few changes:
```julia
ablines([],[]) # fails
arrows(Point3f[], Vec3f[]) # fails
arrows2d(Point2f[], Vec2f[]) # fails
arrows3d(Point3f[], Vec3f[]) # fails
scatter(Point3f[]) # works
band(Point3f[],Point3f[]) # works
barplot(Float64[],Float64[]) # works 
boxplot(Float64[],Float64[]) # works
lines(Point3f[]) # works
linesegments(Point3f[]) # works
meshscatter(Point3f[]) # works
pie(Float64[]) # works
scatterlines(Point3f[]) # works
triplot(Point3f[]) # fails deliberately
```

Differencing with the previous status (July 2025 vs Aug 2024):
- `ablines` fails with `reducing over an empty collection is not allowed`
- `arrows` (deprecated) had worked but now fails with `Conversion failed for MeshScatter`
- `arrows2d`/`arrows3d` have replaced `arrows` but fail in the same way
- `band` now works
- *(possibly the same as before)* `triplot` fails with `InsufficientPointsError`, so it's not supposed to work

So `ablines` and `arrows`/`arrows2d`/`arrows3d` are the ones remaining to fix.
